python_modules/libraries/dagster-dbt/dagster_dbt/dbt_artifacts.py

```diff
@@ -0,0 +1,134 @@
+import os
+import shutil
+from functools import cached_property
+from pathlib import Path
+from typing import List, Optional, Union
+
+from .core.resources_v2 import DbtCliResource
+from .errors import DagsterDbtManifestNotPreparedError
+
+
+def _should_compile() -> bool:
+    return (
+        # if launched via `dagster dev` cli
+        bool(os.getenv("DAGSTER_IS_DEV_CLI"))
+        or
+        # or if explicitly opted in
+        bool(os.getenv("DAGSTER_DBT_PARSE_PROJECT_ON_LOAD"))
+    )
+
+
+class DbtArtifacts:
+    def __init__(
+        self,
+        project_dir: Union[Path, str],
+        *,
+        target_folder: str = "target",
+        prepare_command: List[str] = ["parse", "--quiet"],
+        package_data_dir: Optional[Union[Path, str]] = None,
+    ):
+        """A utility class to help manage dbt artifacts in different deployment contexts.
+
+        This class provides a setup to solve for these goals:
+        * During development, reload the manifest during start-up to pick up any changes.
+        * When deploying, use a compiled manifest to reduce start-up time.
+        * Handle the scenario when the dbt project is copied in to a directory for packaging.
+
+        Args:
+            project_dir (Union[str, Path]):
+                The directory of the dbt project.
+            target_folder (Union[str, Path]):
+                The folder in the project project directory to output artifacts.
+                Default: "target"
+            prepare_command: The dbt cli command to run to prepare the manifest.json
+                Default: ["parse", "--quiet"]
+
+        """
+        self.project_dir = project_dir if isinstance(project_dir, Path) else Path(project_dir)
+        self._target_folder = (
+            target_folder if isinstance(target_folder, Path) else Path(target_folder)
+        )
+        self._prepare_command = prepare_command
+
+        if package_data_dir is None:
+            self._package_data_dir = None
+        elif isinstance(package_data_dir, Path):
+            self._package_data_dir = package_data_dir
+        else:
+            self._package_data_dir = Path(package_data_dir)
+
+    @cached_property  # to prevent repeat compilation
+    def manifest_path(self) -> Path:
+        """The path to the manifest.json, compiling the manifest first if in development or
+        ensuring it already exists if not.
+        """
+        if _should_compile():
+            return self._prepare_manifest()
+
+        p = self._base_dir / self._target_folder / "manifest.json"
+        if not p.exists():
+            raise DagsterDbtManifestNotPreparedError(
+                f"Did not find prepared manifest.json at expected path {p}.\n"
+                "If this is an environment that is expected to prepare the manifest on load, "
```

Comment:
We use "run time" and "build time" in our reference https://docs.dagster.io/integrations/dbt/reference#loading-dbt-models-from-a-dbt-project (even though our environment variable is `XXX_ON_LOAD`, sorry!).

Could we use "run time" here?